README.md

```diff
@@ -124,15 +126,79 @@ flyway -user=$USER -password="" -locations="filesystem:database/sql/" -url=jdbc:
 
 ### Local Test Network
 
-With a testnet (e.g. [Ganache](https://www.trufflesuite.com/ganache)) running on `localhost:8545` deploy the contracts via:
+In order to run the `e2e` tests you have to have a testnet running locally.
+Due to the RPC calls the services issue `Ganache` is incompatible, so we will use `hardhat`.
+
+1. Install [npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm)  
+2. Install hardhat with `npm install --save-dev hardhat`  
+3. Create `hardhat.config.js` in the directory you installed `hardhat` in with following content:
+<pre><code>module.exports = {
+        networks: { 
+            hardhat: {
+                initialBaseFeePerGas: 0,
+                accounts: {
+                    accountsBalance: "1000000000000000000000000"
+                }
+            }
+        }
+};</code></pre>
+4. Run local testnet with `npx hardhat node`
+
+## Running the Services Locally
+
+### Prerequisites
+Reading the state of the blockchain requires issuing RPC calls to an ethereum node. This can be a testnet you are running locally, some "real" node you have access to or the most convenient thing is to use a third party service like [infura](https://infura.io/) to get access to an ethereum node which we recommend.
+After you made a free infura account they offer you "endpoints" for the mainnet and different testnets. We will refer those as `node-urls`.
+Because Gnosis only runs their services on mainnet, rinkeby and gnosis chain you need to select one of those.
+
+Note that the `node-url` is sensitive data. The `orderbook` and `solver` executables allow you to pass it with the `--node-url` parameter. This is very convenient for our examples but to minimize the possibility of sharing this information by accident you should consider setting the `NODE_URL` environment variable so you don't have to pass the `--node-url` argument to the executables.
+
+To avoid confusion during your tests, always double check that the token and account addresses you use actually correspond to the network of the `node-url` you are running the executables with.
+
+### Orderbook
+
+To see all supported command line arguments run `cargo run --bin orderbook -- --help`.
+
+Run an `orderbook` on `localhost:8080` with:
+
+```sh
+cargo run --bin orderbook -- \
+  --skip-trace-api true \
```

Comment:
Also useful would be to add `--skip-event-sync` which skips the syncing of past `GPv2Settlement` contract events and would speed up the startup for running locally.